pcawg_metadata_parser/compare_xml_md5sum.py

```diff
@@ -0,0 +1,250 @@
+    md5sum.append(hashlib.md5(effective_eq_xml).hexdigest())
+
+    # take out the qc_metrics section
+    analysis_attrib = get_analysis_attrib(gnos_analysis)
+    qc_metrics_xml = json.dumps(json.loads(analysis_attrib.get('qc_metrics')) if analysis_attrib.get('qc_metrics') else [], indent=4, sort_keys=True)
+    md5sum.append(hashlib.md5(qc_metrics_xml).hexdigest())
+
+    # take out the different file_type section
+    for file_type in ['.bam', '.bai']:
+        file_xml_dict = get_file_info(gnos_analysis.get('files').get('file'), file_type)
+        if file_xml_dict: 
+            file_xml = json.dumps(file_xml_dict, indent=4, sort_keys=True)
+            md5sum.append(hashlib.md5(file_xml).hexdigest())
+        else:
+            md5sum.append('missing')
+
+    if xml_dir: 
+        with open(xml_dir + '/' + fname +'_ordered_'+get_formal_repo_name(repo), 'w') as y:
+            y.write(effective_eq_xml)
+
+    return md5sum
+
+
+def get_gnos_repo(gnos_repo, source_gnos_repo, target_gnos_repo):
+    if not source_gnos_repo or not target_gnos_repo:
+        return gnos_repo
+
+    if source_gnos_repo in gnos_repo and target_gnos_repo in gnos_repo:
+        gnos_repo = [source_gnos_repo, target_gnos_repo]
+    else:
+        gnos_repo = []
+
+    return gnos_repo 
+
+
+def main(argv=None):
+
+    parser = ArgumentParser(description="Compare the effective xml md5sum among repos",
+             formatter_class=RawDescriptionHelpFormatter)
+    parser.add_argument("-m", "--metadata_dir", dest="metadata_dir",
+             help="Directory containing metadata manifest files", required=True)
+    parser.add_argument("-f", "--gnos_id_file", dest="fname",
+             help="Specify gnos_id file to process", required=True)
+    parser.add_argument("-s", "--source_gnos_repo", dest="source_gnos_repo",
+             help="Specify source_gnos_repo to do the comparation", required=False)
+    parser.add_argument("-t", "--target_gnos_repo", dest="target_gnos_repo",
+             help="Specify target_gnos_repo to do the comparation", required=False)
+    parser.add_argument("-o", "--ordered_xml output folder", dest="xml_dir",
+             help="Specify output folder for the ordered_xml", required=False)
+    parser.add_argument("-i", "--Result of the comparation", dest="compare_result",
+             help="Specify file for the compare result", required=False)
+    parser.add_argument("-u", "--Whether download metadata xml", dest="download_xml",
+             help="Specify whether download_metadata_xml or not", required=False)
+
+    args = parser.parse_args()
+    metadata_dir = args.metadata_dir  # this dir contains gnos manifest files, will also host all reports
+    fname = args.fname
+    source_gnos_repo = get_formal_repo_name(args.source_gnos_repo) if args.source_gnos_repo else None
+    target_gnos_repo = get_formal_repo_name(args.target_gnos_repo) if args.source_gnos_repo else None
```

Comment:
why need to choose source/target repo? shouldn't it be automatically report all repos that are involved in any md5sum differences?